packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -642,6 +642,13 @@ export interface BucketProps {
    * @see https://docs.aws.amazon.com/AmazonS3/latest/dev/access-control-block-public-access.html
    */
   readonly blockPublicAccess?: BlockPublicAccess;
+
+  /**
+   * The metrics configuration of this bucket.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-metricsconfiguration.html
+   */
+  readonly metricsConfigurations?: Array<CfnBucket.MetricsConfigurationProperty | cdk.Token> | cdk.Token;
```

Comment:
We don't expose L1 property types at the L2 layer. Please define a mirror of this structure and see what (if any) properties of it can be made optional by deriving sane defaults.

Also please don't expose type unions.

Don't forget to add a unit test.